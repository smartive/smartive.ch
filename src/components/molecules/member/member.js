import PropTypes from 'prop-types';
import React from 'react';
import Img from 'gatsby-image';

import './member.scss';
import NoPortraitYet from './no-portrait-yet.inline.svg';

export const Member = ({ name, job, image, education, children, links = [] }) => (
  <div className="member col-xs-12 col-md-6 col-lg-6" itemScope itemProp="employee" itemType="http://schema.org/Person">
    {image ? <Img className="member__portrait" fluid={image.fluid} alt={name} /> : <NoPortraitYet />}
    <div className="member__text">
      <h2>
        <small itemProp="jobTitle">{job}</small> <span itemProp="name">{name}</span>
      </h2>
      <h3>{education}</h3>
      {children}
      <p>
        {links.length > 0
          ? links
              .map((link) => {
                const linkItemProp = link.url.match(/^mailto:.+$/i) ? 'email' : 'sameAs';
                return (
                  <a key={link.url} href={link.url} itemProp={linkItemProp}>
                    {link.text}
                  </a>
                );
              })
              .reduce((pre, cur) => [pre, ', ', cur])
          : null}
      </p>
    </div>
  </div>
);

Member.propTypes = {
  name: PropTypes.string.isRequired,
  job: PropTypes.string.isRequired,
  image: PropTypes.shape({
    fluid: PropTypes.shape({
      base64: PropTypes.string,
      aspectRatio: PropTypes.number,
      src: PropTypes.string,
      srcSet: PropTypes.string,
      srcSetType: PropTypes.string,
      sizes: PropTypes.string,
      originalImg: PropTypes.string,
    }),
  }),
  education: PropTypes.string.isRequired,
  children: PropTypes.element.isRequired,
  links: PropTypes.arrayOf(
    PropTypes.shape({
      url: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
    })
  ),
};

Member.defaultProps = {
  links: [],
  image: null,
};

export default Member;
