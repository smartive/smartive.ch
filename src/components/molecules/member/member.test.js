import React from 'react';
import renderer from 'react-test-renderer';

import { Member } from './member';

describe('Member', () => {
  it('renders without links', () => {
    const tree = renderer
      .create(
        <Member name="Foo Bar" job="Bazzer" image={null} education="Funny dude">
          <p>Content</p>
        </Member>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with links', () => {
    const tree = renderer
      .create(
        <Member
          name="Foo Bar"
          job="Bazzer"
          image={null}
          education="Funny dude"
          links={[
            { url: '/foo', text: 'Foo' },
            { url: '/bar', text: 'Bar' },
          ]}
        >
          <p>Content</p>
        </Member>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
