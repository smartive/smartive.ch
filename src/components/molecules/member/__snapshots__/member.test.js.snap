// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member renders with links 1`] = `
<div
  className="member col-xs-12 col-md-6 col-lg-6"
  itemProp="employee"
  itemScope={true}
  itemType="http://schema.org/Person"
>
  <test-file-stub />
  <div
    className="member__text"
  >
    <h2>
      <small
        itemProp="jobTitle"
      >
        Bazzer
      </small>
       
      <span
        itemProp="name"
      >
        Foo Bar
      </span>
    </h2>
    <h3>
      Funny dude
    </h3>
    <p>
      Content
    </p>
    <p>
      <a
        href="/foo"
        itemProp="sameAs"
      >
        Foo
      </a>
      , 
      <a
        href="/bar"
        itemProp="sameAs"
      >
        Bar
      </a>
    </p>
  </div>
</div>
`;

exports[`Member renders without links 1`] = `
<div
  className="member col-xs-12 col-md-6 col-lg-6"
  itemProp="employee"
  itemScope={true}
  itemType="http://schema.org/Person"
>
  <test-file-stub />
  <div
    className="member__text"
  >
    <h2>
      <small
        itemProp="jobTitle"
      >
        Bazzer
      </small>
       
      <span
        itemProp="name"
      >
        Foo Bar
      </span>
    </h2>
    <h3>
      Funny dude
    </h3>
    <p>
      Content
    </p>
    <p />
  </div>
</div>
`;
